shell/server/drivers/ip.js

```diff
@@ -290,35 +307,49 @@ Meteor.startup(() => {
 });
 
 IpRemoteHostImpl = class IpRemoteHostImpl {
-  constructor(address) {
+  constructor(address, tls) {
     if (address.upper64 || address.upper64 === 0) {
       // address is an ip.capnp:IpAddress, we need to convert it
       this.address = addressToString(address);
     } else {
       this.address = address;
     }
+
+    this.tls = tls;
   }
 
   getTcpPort(portNum) {
-    return { port: new TcpPortImpl(this.address, portNum) };
+    return { port: new TcpPortImpl(this.address, portNum, this.tls) };
   }
 
   getUdpPort(portNum) {
+    if (this.tls) {
+      const error = new Error("UDP over TLS is not yet supported");
```

Comment:
Technically this would be "TLS over UDP", or perhaps more precisely "DTLS over UDP", rather than "UDP over TLS".